### PR TITLE
pg_restore/pg_dump: add page

### PR DESCRIPTION
**Pull request by mz026**

### CODE REVIEW

File: pages/common/pg_restore.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# pg\_restore
```
Comment (igorshubovych) [inline pages/common/pg_restore.md]: why do write it like this?
```md
# pg\_restore
```

it has to be 
```md
# pg_restore
```

File: pages/common/pg_restore.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+> Restore a PostgreSQL database from an archive file created by pg_dump
+
+- restore an archive into an existed database
+
+`pg_restore --no-owner -h {{localhost}} -p {{5432}} -U {{username}} -d {{db_name}} {{archive_file.dump}}`
```
Comment (igorshubovych) [inline pages/common/pg_restore.md]: Check out [Contributing Guidelines, rule number 4](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines)

> Introduce options gradually, starting with the simplest commands and using more complex examples progressively.

Comment (mz026) [inline pages/common/pg_restore.md]: Thanks for pointing out! I updated the page :)

Comment (mz026) [inline pages/common/pg_restore.md]: updated, thanks!

2016-01-07 17:18 GMT+08:00 Igor Shubovych <notifications@github.com>:

File: pages/common/pg_restore.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+> Restore a PostgreSQL database from an archive file created by pg_dump
+
+- restore an archive into an existed database
+
+`pg_restore -d {{db_name}} {{archive_file.dump}}`
+
+- customize host, port, username
+
+`pg_restore -h {{localhost}} -p {{5432}} -U {{username}} -d {{db_name}} {{archive_file.dump}}`
```
Comment (igorshubovych) [inline pages/common/pg_restore.md]: localhost it is a alias for concrete address.
In the example it should be one of two:

```md
`pg_restore -h {{host}}`
`pg_restore localhost`
```

File: pages/common/pg_restore.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+- not require ownership of objects to match the original database
```
Comment (igorshubovych) [inline pages/common/pg_restore.md]: i don't think `--no-owner` option should be in tldr. It is not a trivial thing. Not for newbies. I would rather split the previous example to several examples for each option.

**Comment by igorshubovych** [comment]
```md
`pg_restore -h {{localhost}} -p {{5432}} -U {{username}} -d {{db_name}} {{archive_file.dump}}`
```
This is complicated example, and you introduce 5 options in it. It is against tldr guidelines. Please split it.

Take a look at [mysql page](https://github.com/tldr-pages/tldr/blob/master/pages/common/mysql.md) or

**Comment by mz026** [comment]
just updated, thanks for reviewing!!

2016-01-07 21:15 GMT+08:00 Igor Shubovych <notifications@github.com>:

**Comment by igorshubovych** [comment]
I like it much more! Thanks. LGTM.

**Comment by igorshubovych** [comment]
@mz026 Could you plz squash the commits?

**Comment by mz026** [comment]
squashed, thanks! @igorshubovych

File: pages/common/pg_restore.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# pg_restore
+
+> Restore a PostgreSQL database from an archive file created by pg_dump
+
+- restore an archive into an existed database
```
Comment (waldyrious) [inline pages/common/pg_restore.md]: existed --> existing

File: pages/common/pg_restore.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+`pg_restore -d {{db_name}} {{archive_file.dump}}`
+
+- customize username
```
Comment (waldyrious) [inline pages/common/pg_restore.md]: Maybe "same as above, with custom username"?

File: pages/common/pg_restore.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+`pg_restore -U {{username}} -d {{db_name}} {{archive_file.dump}}`
+
+- customize host and port
```
Comment (waldyrious) [inline pages/common/pg_restore.md]: Here too.

File: pages/common/pg_restore.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+`pg_restore -h {{host}} -p {{port}} -d {{db_name}} {{archive_file.dump}}`
+
+- clean database object before creating them
```
Comment (waldyrious) [inline pages/common/pg_restore.md]: What is "them" here? This description isn't clear to me.

Comment (mz026) [inline pages/common/pg_restore.md]: Updated the above. For the "them" meaning, the "database object" should be plural, and the "them" means the "database objects".

**Comment by notpeter** [comment]
You wanna do one for pg_dump too?

**Comment by mz026** [comment]
sure! I added a `pg_dump` page. @notpeter 
Should I separate them into 2 PRs?

**Comment by notpeter** [comment]
I'm not the repo owner but I'm sure one PR for paired commands like pg_dump & pg_restore is fine.

**Comment by mz026** [comment]
haha! I'll separate them then.

**Comment by waldyrious** [comment]
@mz026 I get the feeling you misunderstood @notpeter. Indeed a single PR for related commands is quite acceptable especially since it's desirable to keep any suggested changes in sync. See for instance #605.

**Comment by mz026** [comment]
thanks for clarifying @waldyrious . I added a `pg_dump` page on it :smile: @notpeter

**Comment by igorshubovych** [comment]
It is almost good.
Could you please tune markup a little bit according to our recent changes in [guidelines](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md#markdown-format). Capital letters in the beginning and colons in the end of examples' descriptions.

**Comment by mz026** [comment]
Just updated, thanks! @igorshubovych

**Comment by igorshubovych** [comment]
Thank you for contribution! :+1:

**Pull request merged**